core/test/integration/synchronization/ethereum_synchronization.cpp

```diff
@@ -113,12 +117,24 @@ TEST_F(EthereumLikeWalletSynchronization, MediumXpubSynchronization) {
                     auto tx = wait(txBuilder->build());
                     EXPECT_EQ(tx->getValue()->toLong(), balance->toLong() - (gasLimit * gasPrice));
 
+                    auto now = std::time(nullptr);
+                    char now_str[256];
+                    std::strftime(now_str, sizeof(now_str), "%Y-%m-%dT%H:%M:%SZ", std::localtime(&now));
+                    std::cout << now_str << std::endl;
+
+                    auto history = wait(account->getBalanceHistory(
+                        "2019-09-20T00:00:00Z",
+                        now_str,
+                        api::TimePeriod::DAY
+                    ));
+
+                    EXPECT_EQ(history.back()->toString(), balanceStr);
```

Comment:
[OPT] weak test :-p

Reply:
What do you mean weak test? I don’t really see what else we can do since we don’t know any of the previous balances but the last one. :)

Reply:
Took the address which had a problem (ping @malikfassi ), and check each value that was wrong before and is correct now.

Reply:
I think it was a private-like xpub we cannot put in our integration tests IIRC.